presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayLessThanOperator.java

```diff
@@ -72,6 +56,45 @@ public static boolean lessThan(MethodHandle lessThanFunction, Type type, Block l
                 if ((boolean) lessThanFunction.invoke(rightElement, leftElement)) {
                     return false;
                 }
+                if (!(boolean) equalsFunction.invoke(leftElement, rightElement)) {
+                    return false;
+                }
+            }
+            catch (Throwable t) {
+                Throwables.propagateIfInstanceOf(t, Error.class);
+                Throwables.propagateIfInstanceOf(t, PrestoException.class);
+
+                throw new PrestoException(INTERNAL_ERROR, t);
+            }
+            index++;
+        }
+
+        return leftArray.getPositionCount() < rightArray.getPositionCount();
+    }
+
+    @TypeParameter("T")
+    @SqlType(StandardTypes.BOOLEAN)
+    @TypeParameterSpecialization(name = "T", nativeContainerType = long.class)
+    public static boolean lessThanLong(
+            @OperatorDependency(operator = LESS_THAN, returnType = StandardTypes.BOOLEAN, argumentTypes = {"T", "T"}) MethodHandle lessThanFunction,
+            @TypeParameter("T") Type type,
+            @SqlType("array(T)") Block leftArray,
+            @SqlType("array(T)") Block rightArray)
+    {
+        int len = Math.min(leftArray.getPositionCount(), rightArray.getPositionCount());
+        int index = 0;
+        while (index < len) {
+            checkElementNotNull(leftArray.isNull(index), ARRAY_NULL_ELEMENT_MSG);
+            checkElementNotNull(rightArray.isNull(index), ARRAY_NULL_ELEMENT_MSG);
+            long leftElement = type.getLong(leftArray, index);
+            long rightElement = type.getLong(rightArray, index);
+            try {
+                if ((boolean) lessThanFunction.invokeExact(leftElement, rightElement)) {
+                    return true;
+                }
+                if ((boolean) lessThanFunction.invokeExact(rightElement, leftElement)) {
```

Comment:
This is still incorrect. "long" is just the internal representation for the values of type T. It's not necessarily for representing "BIGINT" or any other integral type. There might be a type that doesn't have a total order that uses long as the internal representation.